doc/rm/asm_coding_style.md

```diff
@@ -0,0 +1,275 @@
+---
+title: "RISC-V Assembly Style Guide"
+---
+
+## Basics
+
+### Summary
+
+OpenTitan needs to implement substantial functionality directly in RISC-V assembly.
+This document describes best practices for both assembly `.S` files and inline assembly statements in C and C++.
+It also codifies otherwise unwritten style guidelines in one central location.
+
+This document is not an introduction to RISC-V assembly; for that purpose, see the [RISC-V Assembly Programmer's Manual](https://github.com/riscv/riscv-asm-manual/blob/master/riscv-asm.md).
+
+Assembly is typically very specialized; the following rules do not presume to describe every use-case, so use your best judgement.
+
+This style guide is specialized for R32IMC, the ISA implemented by Ibex. 
+As such, no advice is provided for other RISC-V extensions, though this style guide is written such that advice for other extensions could be added without conflicts.
+
```

Comment:
The style guide should also include the "Terminology Conventions" sections as present in the other style guides.

Reply:
Funny enough, it originally did, and I was asked to remove it. I think it was either @imphil or @asb? I might be misremembering.

Reply:
Ok, no worries.